torch/_inductor/codegen/cpp_template_kernel.py

```diff
@@ -240,49 +244,98 @@ def store_pointwise_nodes(
         if not reindexers:
             reindexers = [None] * len(nodes)
         assert len(offsets) == len(var_sizes[0])
-        output_index = dst.get_layout().make_indexer()([*var_ranges.keys()])
+        output_index = ref_dst.get_layout().make_indexer()([*var_ranges.keys()])
         kernel_group = KernelGroup()
         kernel_group.args = self.args
         cpp_kernel_proxy = CppKernelProxy(kernel_group)
         bodies = []
         var_sizes_list = []
-        for i, node in enumerate(nodes):
-            output_name = node.get_name() if i < len(nodes) - 1 else dst.get_name()
-            node = node.data if isinstance(node, ir.ComputedBuffer) else node
-            assert isinstance(node, ir.Pointwise), node
-
-            def fn(*args):
-                assert len(args) == 2
-                assert len(args[0]) == len(var_sizes[0])
-                assert len(args[1]) == 0
-                new_args = [arg + offset for arg, offset in zip(args[0], offsets)]  # type: ignore[arg-type]
-                if reindexers[i] is not None:
-                    new_args = reindexers[i](new_args)  # type: ignore[misc]
-                V.ops.store(
-                    output_name,
-                    output_index,
-                    node.make_loader()(new_args).value,
+        if isinstance(nodes[0], Iterable):
+            # TODO: support for different length of epilogue
+            group_gemm_number = len(nodes)
```

Comment:
Why are they named "gemm"? Is the logic gemm specific? Or we are just handling multiple outputs here? Also, is it possible to share code or extend the implementation without adding an extra code path here?

Reply:
Thanks for the comment. This has been refactored in the follow-up PR enabling epilogue fusion. Specifically:

- The implementation has been split into `store_output` and `store_outputs` in the subsequent PR. If necessary, I can move these changes into the current PR.

- After the refactor, there is no `group_gemm_number` in the epilogues. The epilogues are flattened into a list and code-generated without specifying which GEMM they belong to.